client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -57,25 +58,21 @@ const Checkout = React.createClass( {
 			return;
 		}
 
-		if ( this.props.cart.hasLoadedFromServer ) {
-			this.trackPageView();
-
-			if ( this.props.product ) {
-				this.addProductToCart();
-			}
+		if ( this.props.cart.hasLoadedFromServer && this.props.product ) {
+			this.addProductToCart();
 		}
 
 		window.scrollTo( 0, 0 );
 	},
 
 	componentWillReceiveProps: function( nextProps ) {
-		if ( ! this.props.cart.hasLoadedFromServer && nextProps.cart.hasLoadedFromServer ) {
-			// if the cart hadn't loaded when this mounted, record the page view when it loads
-			this.trackPageView( nextProps );
+		if ( ! this.props.cart.hasLoadedFromServer && nextProps.cart.hasLoadedFromServer && this.props.product ) {
+			this.addProductToCart();
+		}
 
-			if ( this.props.product ) {
-				this.addProductToCart();
-			}
+		if ( this.props.cart.hasPendingServerUpdates && ! nextProps.cart.hasPendingServerUpdates ) {
+			// We only want to track the page view when the cart has finished loading the products
+			this.trackPageView( nextProps );
```

Comment:
Good catch. This behavior should be fine; it will ensure that we're always updating Criteo when the cart changes which seems ideal.